### PR TITLE
feat(3420): privacy mode toggle

### PR DESCRIPTION
**Pull request by vinnyhoward**
## **Description**

Adding in the feature to allow user's to hide their token balance and asset amounts. Here is the [Figma](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?m=auto&node-id=3818-72888&t=KcsTgk5AH10ngOdn-1) for reference

## **Related issues**

Feature: [#3240](https://github.com/MetaMask/MetaMask-planning/issues/3420)

## **Manual testing steps**

1. Go to wallet home page
2. Click on the eye icon/balance
3. Witness your assets and balance turn into dots
4. Magic✨
5. Also check your other accounts in the account selector. The balance and asset amount should be hidden as well

## **Screenshots/Recordings**

priv_mode

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Texts/SensitiveText/SensitiveText.types.ts [hunk 1]
```diff
@@ -42,5 +43,5 @@ export interface SensitiveTextProps extends TextProps {
   /**
    * The text content to be displayed or hidden.
    */
-  children: string;
+  children: string | React.ReactNode;
```
Comment (georgewrmarshall) [inline app/component-library/components/Texts/SensitiveText/SensitiveText.types.ts:46]: nit: I believe `React.ReactNode` is a catch-all type for anything that can be rendered: elements, strings, numbers, etc so we could replace string here

**Review by georgewrmarshall — DISMISSED** [review]
Approving for changes to component-library folder

**Review by gambinish — DISMISSED** [review]
LGTM. Looks like there are some merge conflicts, so jlmk when addressed and I can re-approve

File: app/component-library/components/Texts/SensitiveText/SensitiveText.types.ts (hunk 1 above)
Comment (vinnyhoward) [reply app/component-library/components/Texts/SensitiveText/SensitiveText.types.ts]: Just fixed :D

**Review by gambinish — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by Cal-L — COMMENTED** [review]
LGTM

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**